pyccel/codegen/printing/cwrappercode.py

```diff
@@ -172,68 +374,40 @@ def get_PyArgParseType(self, used_names, variable, argument):
         variable : Variable
             The variable which will be passed to the translated function
 
-        argument : Variable
-            The original argument provided to the function
-
         Returns
         -------
         collect_var : Variable
             The variable which will be used to collect the argument
 
-        cast_func_stmts : list
-            A list of statements to be carried out after parsing the arguments.
-            These handle casting collect_var to variable if necessary
+        cast_fun : FunctionCall
+            call to cast function responsible of the conversion of one data type into another
         """
-        body = []
+        cast_function = None
         collect_var = variable
 
-        if variable.is_optional:
+        if variable.rank > 0:
+            collect_type = PyccelPyArrayObject()
+            collect_var = Variable(dtype= collect_type, is_pointer = True, rank = variable.rank,
+                                    order= variable.order, name=self.get_new_name(used_names, variable.name+"_tmp"))
```

Comment:
Does this need to be after `is_optional`? I don't think I tested an optional array

Reply:
Hmm, I guess it might work if a `PyArrayObject` can be set to `PyNone`. But I'm not sure that's possible. We should test. But we'll have to add whatever function is needed to case a `PyArray` to a `PyArrayObject` if it doesn't work :/

Reply:
1 - PyArrayObject can't be set to PyNone.
2 - I can set the default value to NULL  and check it after PyArg_ParseTupleAndKeywords
3 - If I do that if a user call function like this ```func(None)``` it won't work   PyArg_ParseTupleAndKeywords will raise a Type error.
4 - if it's called like this ```func()```I think it will work because the Pyarrayobject won't be touched in  the  PyArg_ParseTuple
and he will keep the default value NULL

Reply:
That sounds right to me. We have never handled passing Py_None anyway